pkg/git/gitclient/gitclient.go

```diff
@@ -268,11 +270,23 @@ func (g *GitClient) Branch(name string) error {
 
 func (g *GitClient) ValidateRemoteExists(ctx context.Context) error {
 	logger.V(3).Info("Validating git setup", "repoUrl", g.RepoUrl)
-	remote := g.Client.NewRemote(g.RepoUrl, gogit.DefaultRemoteName)
-	// Check if we are able to make a connection to the remote by attempting to list refs
-	_, err := g.Client.ListWithContext(ctx, remote, g.Auth)
-	if err != nil {
-		return fmt.Errorf("connecting with remote %v for repository: %v", gogit.DefaultRemoteName, err)
+
+	repositoryURL, _ := url.Parse(g.RepoUrl)
+	if strings.Contains(repositoryURL.Hostname(), "git-codecommit") && strings.Contains(repositoryURL.Hostname(), "amazonaws.com") {
```

Comment:
is it worth it to move this 'is the url codecommit' logic to a function and the `git-codecommit` and `amazonaws.com` values to constants in their own file (e.g. `pkg/git/codecommit` or w/e) since we're re-using them in a couple places?